server/server.js

```diff
@@ -56,10 +62,55 @@ Server.prototype.terminate = function(done) {
   });
 
   this.sentry.stop();
-  this.socketServer.close();
+
+  if (this.socketServer) {
+    this.socketServer.close();  
+  }
+  
   Core.Persistence.disconnect(done);
 };
 
+Server.prototype.filter = function filter (filter) {
+  this._filters.push(filter);
+};
+
+Server.prototype._runFilters = function() {
+  var context = arguments[0],
+      args = [].slice.call(arguments, 1),
+      length = this._filters.length,
+      index = 0,
+      result = true;
+
+  while (index < length) {
+    var filter = this._filters[index];
+    
+    if (filter && filter[context]) {
+      result = filter[context].apply(filter, args);
+    };
+
+    if (!result) {
+      logging.warn('#socket.message - filter halted execution', args);
+      break;
+    }
+
+    index++;
+  }
+
+  return result;
+};
+
+Server.prototype._runPreFilters = function(c, m, o) {
```

Comment:
(c, m, o) should be (client, message, options) ?  (Breaking your own "don't make it cryptic rule?")